Python/thread_pthread.h

```diff
@@ -462,16 +462,27 @@ PyThread_acquire_lock_timed(PyThread_type_lock lock, PY_TIMEOUT_T microseconds,
     }
 
     _PyTime_t deadline = 0;
-    if (timeout > 0 && !intr_flag) {
+    if (timeout > 0
+#ifndef HAVE_SEM_CLOCKWAIT
+        && !intr_flag
+#endif
+        )
+    {
         deadline = _PyTime_GetMonotonicClock() + timeout;
     }
 
     while (1) {
         if (timeout > 0) {
+#ifdef HAVE_SEM_CLOCKWAIT
+            struct timespec abs_timeout;
+            _PyTime_AsTimespec_clamp(deadline, &abs_timeout);
+            status = fix_status(sem_clockwait(thelock, CLOCK_MONOTONIC, &abs_timeout));
```

Comment:
I don't know if sem_clockwait() can fail with ENOSYS if Python is built with a recent glibc and then run with an older glibc or a different kernel version.

Should we attempt to fallback to sem_timedwait() if sem_clockwait() "doesn't work"? How can we determine if sem_clockwait() "doesn't work"? So far, I failed to find a sem_clockwait() documentation or manual page.

Reply:
In case of doubt, I prefer to leave the code as it is, and only implement a fallback once we can test it on a real system.